Update workflow CI, tests & more

catalinaIlabaca opened the pull request:
Se añadieron las carpetas .pytest_cache y .nox al .gitignore, y se incorporó la generación del badge de cobertura en test.sh junto con actualizaciones en setup.cfg. Se refactorizó el código eliminando importaciones innecesarias y ordenándolas alfabéticamente, y se agregó la importación de traceback para mejorar el manejo de errores. Además, se sumaron varias pruebas para aumentar la cobertura, se actualizaron las instrucciones para ejecutar los tests y se añadió el badge de cobertura en la documentación. Finalmente, se ajustó la información de metadatos en la configuración del setup.

catalinaIlabaca committed:
docs: updates instructions to run test & adds coverage badge

catalinaIlabaca committed:
refactor: delete unnecessary imports and use alphabetical order in importations

catalinaIlabaca committed:
refactor: update setup to change meta info

fsologureng commented on .gitignore:
```diff
@@ -37,13 +37,16 @@ pip-delete-this-directory.txt
 coverage.xml
-*,cover
+*.cover
+*.py,cover
```
este parece ser un typo, me parece que no hay carpeta ni archivo `cover`

catalinaIlabaca replied:
yap, se quitara

fsologureng commented on .gitignore:
```diff
@@ -61,4 +64,8 @@ target/
+*.DS_Store
+
+# openedx test files
+test_root/
```
este queda sólo cuando el test sale mal, no debería hacerse una regla para ello

catalinaIlabaca replied:
comprendo lo sacaré

fsologureng commented on .gitignore:
este cambio no es *ci:* si no *chore:*

fsologureng replied:
sigue metido dentro del `ci:`

fsologureng commented on eol_survey/setup.cfg:
este cambio no es *ci:* si no *chore:*

fsologureng commented on eol_survey/eolsurveyconsumer.py:
```diff
@@ -110,7 +109,6 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
             matlab_api_key=None,
         )
         _ = capa_system.i18n.ugettext
-        print(user_state_iterator)
```
esto debería modificar el comportamiento, y si es intencional, debería ser en otro commit o con una glosa que informe del cambio

catalinaIlabaca committed:
fix: adds traceback importation

catalinaIlabaca committed:
test: adds several tests to improve coverage

- adds test to try submit_problem, studio_view render, workbench_scenarios and validate_block
- adds test to render author and student view and generate_report_data method with limit_responses 0
- adds a test for Survey_form_view when the http client user is anonymous and the method is GET
- refactors test by replaces relative import from . import views with direct imports: from .views import EolSurveyView, EolSurveyReportAnalyticsView.
This makes dependencies explicit and reduces unnecessary namespace usage in test files.

catalinaIlabaca committed:
chore: adds pytest_cache and .nox folders to .gitignore

catalinaIlabaca committed:
chore: adds genbadge coverage to test.sh & updates setup.cfg

fsologureng commented on .github/test.sh:
al agregarse la generación del badge, esto deja de ser sólo ci: porque afecta al local. De hecho es necesario para generar la documentación. Lo dejaría como chore: por lo que hablábamos en persona. Esto sirve para lo local (genera el badge), pero también hace un cambio en la forma de ejecutar los tests localmente (no sólo en github, que es nuestro ci)

fsologureng commented on setup.cfg:
Ídem el otro archivo de este commit, esto es un chore porque es un config que como está sólo tiene relevancia para el badge generado.

fsologureng commented on eol_survey/tests.py:
change commit to test:

fsologureng replied:
Adds several test**s** to **improve** coverage

fsologureng replied:
hay un espacio doble en la descripción del commit tb

fsologureng replied:
"adds a test for Survey_form_view when the http client user is anonymous and the method is GET"

fsologureng commented on eol_survey/eolsurveyconsumer.py:
```diff
@@ -1,28 +1,27 @@
+
+# Installed packages (via pip)
+from django.template import Context, Template
 import pkg_resources
```
no quedó con orden alfabético

fsologureng commented on coverage-badge.svg:
"docs: updates instructions to run test & adds coverage badge"

fsologureng reviewed (changes requested):
Principalmente mejorar las descripciones del trabajo hecho

fsologureng commented:
> Se agrega coverage a los test, se ordenan las importaciones no usadas de test, views y eolsurveyconsumer para orden y limpieza del código

@catalinaIlabaca Mejorar descripción del PR porque hace bastante más que lo descrito.
Y no "agrega coverage", aumenta el coverage y agrega el badge de coverage al repo.

fsologureng commented on .github/test.sh:
chore: adds genbadge coverage to test.sh & updates setup.cfg

fsologureng commented:
> > Se agrega coverage a los test, se ordenan las importaciones no usadas de test, views y eolsurveyconsumer para orden y limpieza del código
> 
> @catalinaIlabaca Mejorar descripción del PR porque hace bastante más que lo descrito. Y no "agrega coverage", aumenta el coverage y agrega el badge de coverage al repo.

esto sigue pendiente

fsologureng commented on eol_survey/api.py:
```diff
@@ -1,10 +1,14 @@
+# Internal project dependencies
+from .models import Survey    
+
 import logging
```
desordenado

fsologureng commented on eol_survey/task.py:
```diff
@@ -1,21 +1,29 @@
+from opaque_keys.edx.keys import UsageKey
+
+# Internal project dependencies
+from .utils import get_all_states,_build_student_data, get_all_enrolled_users
```
agregar espacio dp de la coma

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1,35 +1,38 @@
+# Installed packages (via pip)
 from django.test import Client
-from django.urls import reverse
 from django.test import TestCase
```
juntar en 1 linea los 2 imports de django.test

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1,35 +1,38 @@
+# Edx dependencies
+from common.djangoapps.student.roles import CourseInstructorRole, CourseStaffRole
+from common.djangoapps.student.tests.factories import CourseAccessRoleFactory
+from common.djangoapps.student.tests.factories import UserFactory, CourseEnrollmentFactory
```
juntar en 1 linea con el anterior

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1,35 +1,38 @@
-
 
+# Internal project dependencies
+from . import views
```
cambiar por `from .views import EolSurveyView`

jjorgeb replied:
ya que algunas views se estan importando en la linea siguiente, revisar si este import aun es necesario.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -583,6 +596,42 @@ def test_delete_survey_anon(self):
 
+    def test_delete_survey_id_doesnt_exist(self):
+        """
+            when you try to delete a survey, but the survey_id doesn't exist . 
```
Mejorar redacción de la descripción.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -583,6 +596,42 @@ def test_delete_survey_anon(self):
+
+    def test_delete_survey_id_is_empty(self):
+        """
+            when you try to delete a survey, but the survey_id is empty . 
```
Mejorar redacción de la descripción, está casi igual que la anterior.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -958,7 +1012,27 @@ def test_submit_problem_score(self):
+    
+    def test_submit_problem_score_using_current_time(self):
+        """
+            Adds current_time to 
```
Descripción incompleta?

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1040,7 +1113,56 @@ def _get_descriptor_error(self):
+
+    def test_workbench_scenarios(self):
+        """
+            Checks workbench scenarios title and basic scenario
```
Mejorar descripción. No se entiende.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1040,7 +1113,56 @@ def _get_descriptor_error(self):
+    
+    def test_generate_report_data_with_limit_responses_cero(self):
+        """
+            report with limit response using 0 value
```
Mejorar descripción de qué hace o verifica el test.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1408,3 +1530,37 @@ def test_get_eol_survey_responses_no_staff(self):
+        self.assertFalse(response)
+
+    def test__default_data(self):
+        expected_value = {
```
Agregar descripción.

catalinaIlabaca replied:
Revise la función y no se usa, así que la eliminare, mi teoría es que se basaron en eol_contact_form pues ellos usan una función con estos mismos parámetros

catalinaIlabaca committed:
refactor: delete unused _default_data function

jjorgeb commented on README.md:
```diff
@@ -1,5 +1,7 @@
+
 ![https://github.com/eol-uchile/eol_survey/actions](https://github.com/eol-uchile/eol_survey/workflows/Python%20application/badge.svg)
 
 On this Manager of your surveys can create, delete or edit a different surveys, also use this survey in a course like an xblock and finally, generate a report in the instructor view with all the answers of your students.
```
No estoy seguro si queda fuera del scope de este PR, ya que estaba de antes, pero esta descripcion no se entiende mucho.

fsologureng replied:
Sugiero crear un issue para que sea resuelto dp, para que no quede en el aire.
Si @catalinaIlabaca  cacha que no es parte del PR, no debe dejar que quede en el aire y crear el issue, pero @jjorgeb tb puede crear el issue si no ve respuesta. Sean proactivos ambos para no dejar observaciones en el aire. No es fácil que uno vuelva a revisar código entre tanto código, entonces es importante levantar esos issues cuando los ve, aunque sean desechados dp, porque la oportunidad no es fácil que vuelva y un pr cerrado no se suele volver a revisar.

jjorgeb commented on eol_survey/eolsurveyconsumer.py:
```diff
@@ -1,28 +1,28 @@
 from xmodule.util.xmodule_django import add_webpack_to_fragment
-from xblockutils.studio_editable import StudioEditableXBlockMixin
-from xblock.fields import Integer, Scope, String, Dict, Float, Boolean, List, DateTime, JSONField
+from xmodule.x_module import(shim_xmodule_js)
```
Se ve raro este import.

catalinaIlabaca replied:
si, se ve raro, pero estaba de antes

catalinaIlabaca replied:
Lo modifique

jjorgeb commented on eol_survey/tests.py:
```diff
@@ -180,6 +182,18 @@ def test_create_survey_get(self):
+    
+    def test_Survey_form_view_get_anon(self):
+        """
+            Test when Survey_form_view is call by an anonymous client
```
typo: Test when Survey_form_view is called by an anonymous client

jjorgeb commented:
Usar import/imports por sobre importation en los commits.

jjorgeb commented on eol_survey/views.py:
```diff
@@ -1,42 +1,28 @@
-from django.urls import reverse
-from django.conf import settings
-from .task import task_process_data
+from django.http import HttpResponse, JsonResponse
```
Juntar este import con el de arriba.

jjorgeb commented on eol_survey/utils.py:
```diff
@@ -1,19 +1,23 @@
-from django.http import HttpResponse
 from django.contrib.auth.models import User
 from django.core.exceptions import FieldError
+from django.http import HttpResponse
```
En la linea 7 tambien se importa desde django.http.

jjorgeb commented on eol_survey/tests.py:
```diff
@@ -1038,7 +1114,58 @@ def _get_descriptor_error(self):
+    def test_generate_report_data_with_limit_responses_zero(self):
+        """
+            Validate that 'generate_report_data' method correctly handles the optional 
+            'limit_responses' parameter with 0 value
```
with value 0

jjorgeb commented on eol_survey/tests.py:
```diff
@@ -1038,7 +1114,58 @@ def _get_descriptor_error(self):
 
+    def test_workbench_scenarios(self):
+        """
+            Checks that 'workbench_scenarios' methods returns the expected title and XML 
```
En algunos test se use Check y en otros checks, trataria de usar uno solo.

fsologureng commented on eol_survey/api.py:
```diff
@@ -1,11 +1,15 @@
+
+# Installed packages (via pip)
 from django.http import HttpResponse
 from django.http import JsonResponse
```
todavía importaciones desde el mismo módulo en diferentes lineas

fsologureng commented on eol_survey/tests.py:
```diff
@@ -26,10 +25,11 @@
 from xblock.fields import ScopeIds
 from xblock.scorable import Score
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from lms.djangoapps.courseware.models import StudentModule
```
desordenado

fsologureng commented on eol_survey/tests.py:
```diff
@@ -26,10 +25,11 @@
 
 # Internal project dependencies
-from . import views
+from .views import EolSurveyView, EolSurveyReportAnalyticsView
```
Este cambio y sus refactors relacionados abajo, debe ser mencionado en la glosa del commit, sin cambiar el tipo `test:`. El commit no es sólo agregación de tests.

fsologureng commented on eol_survey/tests.py:
```diff
@@ -182,6 +182,18 @@ def test_create_survey_get(self):
         self.assertFalse("Encuesta creada correctamente" in response._container[0].decode())
         self.assertEqual(response.status_code, 200)
+    
+    def test_Survey_form_view_get_anon(self):
```
cambiar Survey por survey en el nombre para seguir la nomenclatura de todos los módulos

fsologureng commented on eol_survey/tests.py:
```diff
@@ -34,7 +35,6 @@
 from .task import  generate
 from eol_survey.models import Survey
 
-import logging
```
este cambio debería ser squasheado con el commit que reordena los imports

fsologureng commented on eol_survey/api.py:
```diff
@@ -3,8 +3,7 @@
 # Installed packages (via pip)
-from django.http import HttpResponse
-from django.http import JsonResponse
+from django.http import HttpResponse, JsonResponse
```
Ídem, squashear con el commit que refactoriza los imports

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1,44 +1,47 @@
+# Installed packages (via pip)
 from django.test import Client
-from django.urls import reverse
 from django.test import TestCase
```
juntar con anterior

catalinaIlabaca replied:
esto ya esta junto en el codigo

fsologureng commented on eol_survey/tests.py:
```diff
@@ -1,44 +1,47 @@
+# Edx dependencies
+from common.djangoapps.student.roles import CourseInstructorRole, CourseStaffRole
+from common.djangoapps.student.tests.factories import CourseAccessRoleFactory
+from common.djangoapps.student.tests.factories import UserFactory, CourseEnrollmentFactory
```
juntar con anterior

fsologureng reviewed (changes requested):
Además de las correcciones al commit de los imports, el último commit debe ser squasheado con el cambio en la doc y en el badge porque no es posible que haya cambiado el coverage sólo por cambiar el setup.py.

Y por favor revisar bien antes de pedir review.
